src/main.rs

```diff
@@ -129,17 +148,67 @@ fn main() -> Result<()> {
             .to_lowercase()
     });
 
+    let worktree = cli.worktree_branch.is_some();
+    let worktree_branch = cli.worktree_branch.unwrap();
```

Comment:
1. Useless use of variables. This reduces the readability of the code because the developer will need to lookup that variable to understand what is going on.
2. You're unwrapping an option without taking a reference first. This effectively takes ownership of the String inside the Option. This probably explains why you want to store a boolean in a variable.

It's very interesting because this is a great example where the developer is fighting against the concepts of the language. In this case Rust offers you a handy type "Option" which is more or less a boolean + an arbitrary type. And here you are destructuring the option to take out the primitives of your option: the boolean and the string (the arbitrary type).

And the whole reason for that is probably because you didn't think about taking the value by reference so you fought with the borrow checker. Then, without realizing it, you obscured the code with variables.

```suggestion
```

Reply:
I probably talk a bit too fast about your code being functional because I think you broke the "non-worktree cargo-temp" by unwrapping the option :grin: this probably panics at runtime.

It's also interesting to see how you used `expect` and justify your unwrapping everywhere else in this PR but here you decided to use `unwrap` like it is obvious this won't panic at runtime despite it is obvious that it will panic since `cli.worktree_branch` can be None.